crates/jstzd/src/task/octez_node.rs

```diff
@@ -0,0 +1,293 @@
+                .unwrap_or("localhost:8732".to_owned()),
+            log_file: self.log_file.take().unwrap_or(PathBuf::from(
+                tempfile::NamedTempFile::new().unwrap().path(),
+            )),
+            options: self.options.take().unwrap_or_default(),
+        })
+    }
+}
+
+#[derive(Default)]
+struct ChildWrapper {
+    inner: Option<Child>,
+}
+
+impl ChildWrapper {
+    pub async fn kill(&mut self) -> anyhow::Result<()> {
+        if let Some(mut v) = self.inner.take() {
+            return Ok(v.kill()?);
+        }
+        Ok(())
+    }
+}
+
+#[async_trait]
+impl AsyncDrop for ChildWrapper {
+    async fn async_drop(&mut self) {
+        let _ = self.kill().await;
+    }
+}
+
+#[derive(Default, Clone)]
+pub struct OctezNode {
+    inner: Arc<RwLock<AsyncDropper<ChildWrapper>>>,
+}
+
+#[async_trait]
+impl Task for OctezNode {
+    type Config = OctezNodeConfig;
+
+    /// Spins up the task with the given config.
+    async fn spawn(config: Self::Config) -> Result<Self> {
+        let node = InnerOctezNode {
+            octez_node_bin: Some(config.binary_path),
+            octez_node_dir: config.data_dir,
+        };
+
+        node.config_init(&config.network, "localhost:8731", &config.rpc_endpoint, 0)?;
+        node.generate_identity()?;
+        Ok(OctezNode {
+            inner: Arc::new(RwLock::new(AsyncDropper::new(ChildWrapper {
+                inner: Some(
+                    node.run(
+                        &File::create(&config.log_file)?,
+                        config
+                            .options
+                            .iter()
+                            .map(|s| s as &str)
+                            .collect::<Vec<&str>>()
+                            .as_slice(),
+                    )?,
+                ),
+            }))),
+        })
+    }
+
+    /// Aborts the running task.
+    async fn kill(&mut self) -> Result<()> {
+        let mut inner = self.inner.write().await;
+        Ok(inner.inner_mut().kill().await?)
+    }
+
+    /// Conducts a health check on the running task.
+    async fn health_check(&self) -> Result<bool> {
+        todo!()
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::{OctezNode, OctezNodeConfigBuilder};
+    use crate::task::Task;
+    use anyhow::Result;
+    use rand::{distributions::Alphanumeric, Rng};
+    use std::{fs::File, path::PathBuf};
+
+    pub struct Child {
+        file_path: Option<String>,
+    }
+
+    impl Child {
+        pub fn kill(&mut self) -> std::io::Result<()> {
+            if self.file_path.is_none() {
+                return Err(std::io::Error::new(
+                    std::io::ErrorKind::Other,
+                    anyhow::anyhow!("failed"),
+                ));
+            }
+            File::create(self.file_path.as_ref().unwrap()).unwrap();
+            Ok(())
+        }
+    }
+
+    pub struct InnerOctezNode {
+        pub octez_node_bin: Option<PathBuf>,
+        pub octez_node_dir: PathBuf,
+    }
```

Comment:
Here, you are mocking Child to test that kill command succeeds/fails when octez node is (not)running. But I think you could test this with no overhead as part of `octez_node_test` in the integration tests  I'm not a fan of mocking the Child and InnerOctezNode in this way.

Reply:
Additionally, you could also expose is_running on `ChildWrapper` to check if the process is running and bubble that to the top. 

```
pub async fn is_running(&mut self) -> anyhow::Result<bool> {
        self.inner
            .as_mut()
            .map(|child| {
                let exit_status = child.try_wait()?;
                Ok(exit_status.is_none())
            })
            .unwrap_or(Ok(false))
    }
```

Reply:
> But I think you could test this with no overhead as part of octez_node_test in the integration tests I'm not a fan of mocking the Child and InnerOctezNode in this way.

Did you miss out a few words?

I feel that having too many external dependencies, e.g. child processes, makes unit tests cumbersome and possibly more flakey. `kill` is covered by the integration test as well. I agree that the mocks are not really meaningful here. We can surely remove them if testing with them don't really make sense, but then there is a check for test coverage, unless we decide to skip it for this. Setting up things like signals to kill processes in unit tests to mimic different scenarios and touch the branches sounds too much to me.

That `is_running` method helps remove the "signal file" thing in these test cases. I've added that and now these tests completely run in memory.

Reply:
I'm in favour of removing the mocks. Integration tests will contribute to the test coverage

Reply:
Okay. I've removed the mocks and unit test cases. I've added new cases to cover the config builder.